hypervisor/arch/x86/guest/vlapic.c

```diff
@@ -552,19 +555,21 @@ static void apicv_basic_accept_intr(struct acrn_vlapic *vlapic, uint32_t vector,
 
 	/* If the interrupt is set, don't try to do it again */
 	if (!bitmap32_test_and_set_lock((uint16_t)(vector & 0x1fU), &irrptr[idx].v)) {
-		/* set tmr if corresponding irr bit changes from 0 to 1 */
+		/* update TMR if interrupt trigger mode has changed */
 		vlapic_set_tmr(vlapic, vector, level);
 		vcpu_make_request(vlapic->vcpu, ACRN_REQUEST_EVENT);
 	}
 }
 
 static void apicv_advanced_accept_intr(struct acrn_vlapic *vlapic, uint32_t vector, bool level)
 {
-	int32_t pending_intr = apicv_set_intr_ready(vlapic, vector);
-
-	vlapic_set_tmr(vlapic, vector, level);
+	if (vlapic_is_tmr_changed(vlapic, vector, level)) {
+		/* kick vCPU off to update TMR and EOI exit bitmap if interrupt trigger mode has changed */
+		kick_vcpu(vlapic->vcpu);
+		vlapic_set_tmr(vlapic, vector, level);
```

Comment:
set_tmr first or kick_vcpu first?

Reply:
On second thought, The new TMR and EOI exit bitmap are only used for the new interrupts. It should be OK to kick vCPU off after TMR and EOI exit bitmap has updated and before new interrupts are updated to PIR.